libs/vertexai/langchain_google_vertexai/embeddings.py

```diff
@@ -188,30 +191,41 @@ def _get_embeddings_with_retry(
         self, texts: List[str], embeddings_type: Optional[str] = None
     ) -> List[List[float]]:
         """Makes a Vertex AI model request with retry logic."""
-
-        errors: List[Type[BaseException]] = [
-            ResourceExhausted,
-            ServiceUnavailable,
-            Aborted,
-            DeadlineExceeded,
-        ]
-        retry_decorator = create_base_retry_decorator(
-            error_types=errors, max_retries=self.max_retries
+        if self.model_type == GoogleEmbeddingModelType.MULTIMODAL:
+            return self._get_multimodal_embeddings_with_retry(texts)
+        return self._get_text_embeddings_with_retry(
+            texts, embeddings_type=embeddings_type
         )
 
-        @retry_decorator
-        def _completion_with_retry(texts_to_process: List[str]) -> Any:
-            if embeddings_type and self.instance["embeddings_task_type_supported"]:
-                requests = [
-                    TextEmbeddingInput(text=t, task_type=embeddings_type)
-                    for t in texts_to_process
-                ]
-            else:
-                requests = texts_to_process
-            embeddings = self.client.get_embeddings(requests)
-            return [embs.values for embs in embeddings]
+    def _get_multimodal_embeddings_with_retry(
+        self, texts: List[str]
+    ) -> List[List[float]]:
+        tasks = []
+        for text in texts:
+            tasks.append(
+                self.instance["task_executor"].submit(
+                    self.instance["get_embeddings_with_retry"],
+                    contextual_text=text,
+                )
+            )
+        if len(tasks) > 0:
+            wait(tasks)
+        embeddings = [task.result().text_embedding for task in tasks]
+        return embeddings
```

Comment:
This only gets the Text embeddings, is that the intended purpose for this method?

Reply:
While text models accept a list of strings for embbeding, multimodal just accepts one each time. I wanted to maintain the user signature which allows to send a list of strings and make them to be embedded in parallel behind.
Adding image and text into the same embedding request will just return the embedding for each of them so I kept them separated for now.